File: programmers/python/level1/p92334.py
# 신고 결과 받기
def solution(id_list, report, k):

    answer = [0 for i in range(len(id_list))]
    noti = {name : [] for name in id_list}

    report = list(set(report))
    
    for r in report:
        splited_report = r.split(' ')
        noti[splited_report[1]].append(splited_report[0]) 

    for n in noti:
        if len(noti[n]) >= k:
            for reporter in noti[n]:
                answer[id_list.index(reporter)] += 1
    
    return answer


id_list = ["muzi", "frodo", "apeach", "neo"]
report = ["muzi frodo","apeach frodo","frodo neo","muzi neo","apeach muzi"]

print(solution(id_list, report, 2))

File: programmers/java/src/main/java/level1/P92334.java
package level1;

import java.util.*;

public class P92334 {
    public int[] solution(String[] id_list, String[] report, int k) {
        int[] answer = new int[id_list.length];
        LinkedHashSet<String> linkedHashSet = new LinkedHashSet<>(Arrays.asList(report));
        String[] reportD = linkedHashSet.toArray(new String[]{});

        HashMap<String, ArrayList<String>> noti = new HashMap<>();
        for (String id : id_list) {
            noti.put(id, new ArrayList<>());
        }

        for (String r : reportD) {
            String[] splitedReport = r.split(" ");
            noti.get(splitedReport[1]).add(splitedReport[0]);
        }

        for (String n : noti.keySet()) {
            if (noti.get(n).size() >= k) {
                for (String reporter : noti.get(n)){
                    answer[(Arrays.asList(id_list).indexOf(reporter))] += 1;
                }
            }
        }

        return answer;
    }
}
